x-pack/plugins/alerting/server/plugin.ts

```diff
@@ -229,6 +235,14 @@ export class AlertingPlugin {
     this.licenseState = new LicenseState(plugins.licensing.license$);
     this.security = plugins.security;
 
+    const useDataStreamForAlerts = !!plugins.serverless;
```

Comment:
**question:** Did we want this determined by the existence of the serverless plugin or via the `xpack.alerting.useDataStreamForAlerts` yml setting? We have the ability to set serverless specific configs inside `config/serverless.yml`.

Reply:
I did originally have a config - may need to remove some remaining bits of it.

To do it right, you're supposed to make it conditional on using the `--serverless` (or whatever) option, so that in theory it can't be used in non-serverless.  In the end, to keep it "safe" (we can't use DS for non-serverless, nor aliases for serverless) for now, basing it off the serverless plugin was easier, but also means we can only do functional / integration testing in the serverless tests.

I also realized lots of our existing FT can't be used directly in serverless, as they do things like mess with spaces, which isn't fully supported in serverless.  If we _could_ have (I thought we might), I might have leaned into the config a bit more, as we could run some of our FT with datastreams, some with alias.  Nope!

I think we could change the way we do this later, if we decide we want it more explicit, without breaking any public "API" (since there is none now).

Reply:
This all makes sense to me, thank you! 👍